### PR TITLE
Add Vagrantfile 

### PR DESCRIPTION
**Pull request by rafael**
###  Desc

Long overdue promise of having a [Vagrantfile](https://www.vagrantup.com/) for Vitess project. The following allows people to quickly set up a Vitess development environment. 

@sougou  - If you think this is good to merge, I can follow up with other PR's to add this to the docs.

I tested this Vagrantfile in a ubuntu machine and in Mac OS. VirtualBox seems to work better in mac OS (the VM runs more smoothly  there and tests/building run faster, but the setup works in both cases).

cc: @dweitzman 

### Usage

1) Install vagrant and vmware/VirtualBox in your system.
2) In the vitess project do:
  ```
  $ vagrant up
  ```
3) Connect to the VM:
  ```
  $ vagrant ssh
  ```
4) Build the project:
  ```
  $ cd $VITESS_WORKSPACE
  $ ./vagrant-scripts/vitess/build.sh 
  ```
5) Run the tests:
  ```
  $ ./vagrant-scripts/vitess/test.sh
  ```

There are also scripts to start/stop a Vitess clusters that have port forwarding setup to the local machine.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  config.vm.network "forwarded_port", guest: 15104, host: 15104 # http
+  config.vm.network "forwarded_port", guest: 16104, host: 16104 # grpc
+
+  # Demo Appp
+  config.vm.network "forwarded_port", guest: 8000, host: 8000 # http
+
+  # If possible, use nfs, this gives a good boost to IO operations in the VM.
+  # if you run into with nfs, just remove this from the synced folder
+
+  config.vm.synced_folder ".", "/vagrant/src/vitess.io/vitess", type: "nfs"
```
Comment (guidoiaquinti) [inline Vagrantfile]: this PR reminds me of something @rafael :) 

FYI: the previous `nfs` bug in Vagrant/Virtualbox -> macOS that didn't allow us to use it it's now fixed in the latest High Sierra release.

**Comment by tirsen** [comment]
Why Vagrant instead of Docker?

**Comment by rafael** [comment]
@tirsen - I'm not sure if this still holds because I haven't tried Docker + Vitess + Mac in more than a year, but I ran into many issues where it was really hard to get the tests passing with Docker. The main problem was that integration tests where too heavy in IO and many tests ended up timing out. So running them locally in docker was too flaky.

File: Vagrantfile (hunk 1 above)
Comment (tirsen) [reply Vagrantfile]: I'm still having issues with nfs, even on High Sierra. :-(

Comment (rafael) [reply Vagrantfile]: Oh really :(  - Which issues? I'm High Sierra, working for me.

Comment (tirsen) [reply Vagrantfile]:
```
mount.nfs: Connection timed out
```

**Pull request merged**